src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -178,10 +222,16 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, model_dir, checkpoint_dir,
         logging.info("Resuming from iteration %s", iteration)
 
     callbacks = []
+    force_checkpoint_dir = None
     callbacks.append(checkpointing.print_checkpointed_evaluation(start_iteration=iteration))
     if checkpoint_dir:
         save_checkpoint = checkpointing.save_checkpoint(checkpoint_dir, start_iteration=iteration)
-        callbacks.append(save_checkpoint)
+    else:
+        force_checkpoint_dir = FORCE_CHECKPOINT_DIR
```

Comment:
why are we reassigning the value here?

Reply:
Changed this design to have early stopping support during regular checkpointing as well.